OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnFaction.cs

```diff
@@ -0,0 +1,74 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Grants a condition while the trait is active.")]
+	class GrantConditionOnFactionInfo : ConditionalTraitInfo
+	{
+		[FieldLoader.Require]
+		[GrantedConditionReference]
+		[Desc("Condition to grant.")]
+		public readonly string Condition = null;
+
+		[Desc("Only grant this condition for certain factions.")]
+		public readonly HashSet<string> Factions = new HashSet<string>();
+
+		[Desc("Should it recheck everything when it is captured?")]
```

Comment:
Am no native speaker, what about `Reevaluate if the condition should still be granted when this actor's owner changes.` though?

Reply:
I have copy pasted that from ProvidesPrerequisite, i think it is better to keep that for consistency, or i can edit both.